src/mappings/utils.ts

```diff
@@ -288,6 +309,59 @@ export async function updateTotalDelegation(
   await delegator.save();
 }
 
+export async function updateIndexerCapacity(
+  address: string,
+  event: FrontierEvmEvent
+): Promise<void> {
+  const indexer = await Indexer.get(address);
+  const delegationId = getDelegationId(address, address);
+  const delegation = await Delegation.get(delegationId);
+  const staking = Staking__factory.connect(
+    STAKING_ADDRESS,
+    new FrontierEthProvider()
+  );
+  const eraManager = EraManager__factory.connect(
+    ERA_MANAGER_ADDRESS,
+    new FrontierEthProvider()
+  );
+
+  const leverageLimit = await staking.indexerLeverageLimit();
+
+  if (indexer) {
+    const indexerStake = delegation?.amount;
+    const indexerTotalStake = indexer?.totalStake;
+
+    const stakeCurr = BigNumber.from(indexerStake?.value.value);
```

Comment:
Please confirm that `BigNumber.from`  accept `undefined`. Seems it will throw error

Reply:
I don't think this condition would occur but I've updated for clarity.